libarchive/archive_read_support_format_iso9660.c

```diff
@@ -773,8 +776,9 @@ isSVD(struct iso9660 *iso9660, const unsigned char *h)
 
 	/* Read Root Directory Record in Volume Descriptor. */
 	p = h + SVD_root_directory_record_offset;
-	if (p[DR_length_offset] != 34)
+	if (!isRootDirectoryRecord(p)) {
```

Comment:
The previous check (whether the root directory record length was 34) is really attempting to tell if we have a valid root directory record.  Let's add a new function just for that purpose.